manifest-generator/generator-services/src/main/java/io/hyscale/generator/services/utils/AgentPortBuilder.java

```diff
@@ -0,0 +1,121 @@
+/**
+ * Copyright 2019 Pramati Prism, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.hyscale.generator.services.utils;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Sets;
+import io.hyscale.commons.exception.HyscaleException;
+import io.hyscale.commons.models.ManifestContext;
+import io.hyscale.commons.utils.NormalizationUtil;
+import io.hyscale.generator.services.constants.ManifestGenConstants;
+import io.hyscale.generator.services.exception.ManifestErrorCodes;
+import io.hyscale.generator.services.model.ManifestResource;
+import io.hyscale.generator.services.plugins.PortsHandler;
+import io.hyscale.plugin.framework.models.ManifestSnippet;
+import io.hyscale.plugin.framework.util.GsonSnippetConvertor;
+import io.hyscale.servicespec.commons.model.service.Agent;
+import io.hyscale.servicespec.commons.model.service.Port;
+import io.hyscale.servicespec.commons.model.service.ServiceSpec;
+import io.kubernetes.client.custom.IntOrString;
+import io.kubernetes.client.openapi.models.V1ContainerPort;
+import io.kubernetes.client.openapi.models.V1ServicePort;
+import org.apache.commons.lang3.BooleanUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+
+@Component
+public class AgentPortBuilder extends AgentHelper implements AgentBuilder {
```

Comment:
We can extract out common functionality between PortHandler and AgentPortBuilder and use that at both places